Date: Sun, 23 Apr 1995 21:45:27 GMT
From: jaggy@purplet.demon.co.uk (Mike Jagdis)
Message-Id: <8176@purplet.demon.co.uk>
To: linux-ibcs2@vger.rutgers.edu
Cc: linux-activists@niksula.hut.fi
Subject: IPC problems
X-Mn-Key: IBCS2
Lines: 44

Strangely, several people seem to have discovered the two bugs in the
IPC code within about 48 hours. I say strangely since these bugs have
been there since december...

  Anyway, here's the official patch for those who might have missed
them.

				Mike


===================================================================
RCS file: /usr/CVS/ibcs/iBCSemul/ipc.c,v
retrieving revision 1.15
diff -u -r1.15 ipc.c
--- 1.15	1995/02/13 10:13:10
+++ ipc.c	1995/04/21 13:27:02
@@ -232,7 +232,7 @@
 					lsemun.buf = &ls;
 					old_fs = get_fs();
 					set_fs (get_ds());
-					retval = SYS (ipc) (SEMCTL, arg1, arg2, cmd, lsemun);
+					retval = SYS (ipc) (SEMCTL, arg1, arg2, cmd, &lsemun);
 					set_fs(old_fs);
 
 					lsem_to_isem(&ls, &is);
@@ -383,7 +383,10 @@
 			 *	and we are IBCS, no raddr value to return
 			 */
 #ifdef IPCCALL
+			old_fs = get_fs();
+			set_fs(get_ds());
 			retval = SYS (ipc) (IPCCALL(1,SHMAT), arg1, arg3, &raddr, (char *) arg2);
+			set_fs(old_fs);
 			if (retval >= 0)
 				retval = raddr;
 #else


-- 
.----------------------------------------------------------------------.
| Mike Jagdis                   | Internet: jaggy@purplet.demon.co.uk  |
| 280, Silverdale Road, Earley, | Voice:    +44 1734 266996            |
| Reading RG6 2NU ENGLAND       | Work:     +44 1734 890403            |
`----------------------------------------------------------------------'
